### PR TITLE
feat: `$ref`-type parameters

### LINKED ISSUE #329: Support `$ref`-type parameters, as well as components.parameters
Followup from #236.

We don’t yet support the following schema:

```yaml
paths:
  '/some/path':
    get:
      parameters:
        - "$ref": "#/components/parameters/since"
        - "$ref": "#/components/parameters/per_page"
        - "$ref": "#/components/parameters/page"
```

Add support for `$ref` parameters as well as add the `components.parameters` object to the final response.

There’s an existing spec which illustrates this: `tests/v3/specs/github.yaml`.

### PR DESCRIPTION
**Pull request by gr2m**
fixes #329

### CODE REVIEW

File: src/v3.ts [hunk 1]
```diff
@@ -166,19 +166,39 @@ export default function generateTypesV3(
...
-            Record<string, OpenAPI3Parameter>
+            Record<string, OpenAPI3Parameter | string>
           > = {};
           responses.parameters.forEach((p) => {
-            if (!allParameters[p.in]) allParameters[p.in] = {};
-            // TODO: handle $ref parameters
-            if (p.name) {
-              allParameters[p.in][p.name] = p;
+            if ("$ref" in p) {
+              // return;
```
Comment (drwpow) [inline src/v3.ts:173]: Remove?

File: tests/v3/expected/github.ts [hunk 2]
```diff
@@ -107,8 +107,10 @@ export interface paths {
      */
     get: {
       parameters: {
-        undefined: {};
         query: {
+          per_page: components["parameters"]["per_page"];
+          page: components["parameters"]["page"];
+          since: components["parameters"]["since"];
```
Comment (drwpow) [inline tests/v3/expected/github.ts:113]: 💯

File: tests/v3/index.test.ts [hunk 3]
```diff
@@ -630,4 +630,61 @@ describe("OpenAPI3 features", () => {
       }`)
     );
   });
+
+  it("$ref-type parameters (#329)", () => {
```
Comment (drwpow) [inline tests/v3/index.test.ts:634]: 💯 Great tests!

**Review by drwpow — APPROVED** [review]
This looks great! I think there’s just a conflict from one of your other PRs.

Thanks for the great addition 🙇🏻

File: src/v3.ts (hunk 1 above)
Comment (gr2m) [reply src/v3.ts:173]: done in https://github.com/drwpow/openapi-typescript/pull/340

**Pull request closed without merging**